lib/internal/streams/writable.js

```diff
@@ -527,9 +530,11 @@ function errorBuffer(state) {
     callback(state.errored ?? new ERR_STREAM_DESTROYED('write'));
   }
 
-  const onfinishCallbacks = state[kOnFinished].splice(0);
-  for (let i = 0; i < onfinishCallbacks.length; i++) {
-    onfinishCallbacks[i](state.errored ?? new ERR_STREAM_DESTROYED('end'));
+  if (state[kOnFinished]) {
+    const onfinishCallbacks = state[kOnFinished].splice(0);
+    for (let i = 0; i < onfinishCallbacks.length; i++) {
+      onfinishCallbacks[i](state.errored ?? new ERR_STREAM_DESTROYED('end'));
```

Comment:
Can we just create the error once? The stack trace will be same for all?

Reply:
This is not a hot path I'm very interested in.